### PR TITLE
YURT-49: Put the pieces in place to make Yurt a pip package AND CLI Tool

### LINKED ISSUE #49: Make Yurt a Python Package/CLI Tool
We can refactor...
- [x] away from using fab commands by using CLI client libraries like Click. (refer to Click docs)
- [x] a way to have yurt installed as a pip package (http://docs.python-guide.org/en/latest/shipping/packaging/)

### PR DESCRIPTION
**Pull request by paolopaolopaolo**
- Remove extra dependencies
- More tweaking for packaging (non-functional)
- More adjustments to setup.py
- Move `yurt.py`
- Fixed issues with packaging, refactored paths in `fabfile/setup.py`
- Refactor paths in `fabfile/add.py`
- Add stuff to `.gitignore`
- Remove argument from `new` submethod and update version
- Re-enable templating
- Add context argument to `new_projects`
- Fix `create_project` command
- Remove `main.command` decorator from `load_orchestration...`
- Move `requirements` to base directory
- Put settings setup in function, make `SETTINGS` lowercase
- Remove `executable` parameter from `pip` module call
- Fix naming conflict
- Clean up bad `pip` call
- Add `deploy` to yurt methods
- Closes #49 

@rmutter, this one's a doozy. Changed around the file organization to better suit packaging, implemented click, and added the boilerplate needed for it to become a pip package. I'm going to want to fix the `README.rst` (currently it is copy-pasted from the `README.md`).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,255 +1,75 @@
...
-
-You should try to run the Django server as well:
 
+## Usage
 ```
-cd /vagrant/<project-name>
-python manage.py runserver 0.0.0.0:8000
+    $ cd new_proj
+    $ yurt add_settings
+    $ (nano/vim/subl) fabric_settings.py
```
Comment (rmutter) [inline README.md]: no emacs love?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
+=============================================
```
Comment (rmutter) [inline README.rst]: should we have either rst or md so it's not duplicated? I'm guessing you had to add one for pypi

File: yurt/fabfile/add.py [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
 from fabric.decorators import task
 from fabric.operations import local
 from fabric.context_managers import lcd
+from fabric.operations import local
 
 from utils import get_project_name_from_repo, generate_printable_string, recursive_file_modify, raw_input_wrapper
+from cli import main
+from setup import YURT_PATH
+
+TEMPLATES_PATH = os.path.join(YURT_PATH, "templates")
```
Comment (rmutter) [inline yurt/fabfile/add.py]: is this before the utils import on purpose?

**Comment by paolopaolopaolo** [comment]
@rmutter, did what I didn't want to do at first, which was rebase this branch on top of the vault changes. If these changes look kosher, I'm ready to squash and merge this PR.

**Comment by rmutter** [comment]
🚢

**Pull request merged**